meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1603 @@
+                    node_texture = node
+                    break
+
+            # Find shader node
+            node_shader = None
+            for node in nodes:
+                if node.type.startswith('BSDF'):
+                    node_shader = node
+                    break
+
+            if texture_name == 'NONE':
+                # Unlink texture node
+                if node_texture is not None:
+                    for link in node_texture.outputs[0].links:
+                        node_tree.links.remove(link)
+
+                    nodes.remove(node_texture)
+
+                    # 3D Viewport still shows previous texture when texture link is removed via script.
+                    # As a workaround we trigger desired viewport update by setting color value.
+                    node_shader.inputs[0].default_value = node_shader.inputs[0].default_value
+            else:
+                if node_texture is None:
+                    node_texture = nodes.new(type="ShaderNodeTexImage")
+
+                if (texture_name == 'UV_GRID') or (texture_name == 'COLOR_GRID'):
+                    if texture_name not in bpy.data.images:
+                        bpy.ops.image.new(name=texture_name, generated_type=texture_name)
+                    image = bpy.data.images[texture_name]
+                else:
+                    if texture_name not in bpy.data.images:
+                        texture_path = os.path.join(PATH, "data", texture_name)
+                        image = bpy.data.images.load(texture_path)
+                    else:
+                        image = bpy.data.images[texture_name]
+
+                node_texture.image = image
+
+                # Link texture node to shader node if not already linked
+                if len(node_texture.outputs[0].links) == 0:
+                    node_tree.links.new(node_texture.outputs[0], node_shader.inputs[0])
+
+        # Switch viewport shading to Material Preview to show texture
+        if bpy.context.space_data:
+            if bpy.context.space_data.type == 'VIEW_3D':
+                bpy.context.space_data.shading.type = 'MATERIAL'
+
+        return {'FINISHED'}
+
+
+class OP_MeasurementsToShape(bpy.types.Operator):
+    bl_idname = "object.measurements_to_shape"
+    bl_label = "Measurements To Shape"
+    bl_description = ("Calculate and set shape parameters for specified measurements")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    betas_regressor_female = None
+    betas_regressor_male = None
+    betas_regressor_neutral = None
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if mesh is active object
+            return ((context.object.type == 'MESH') and (context.object.parent.type == 'ARMATURE'))
+        except: return False
+
+    def execute(self, context):
+        obj = bpy.context.object
+        bpy.ops.object.mode_set(mode='OBJECT')
+
+        if self.betas_regressor_female is None:
```

Comment:
Then here we just load them all at once:

    if not self.betas_regressor: # if it's empty
        for gender in ('male', 'female', 'neutral'): # or even better, iterate over members of a gender enum
            regressor_path = os.path.join(PATH, "data", f"measurements_to_betas_{gender}.json")
            self.betas_regressor[gender] = # etc.